komga-webui/src/views/BookReader.vue

```diff
@@ -857,6 +864,64 @@ export default Vue.extend({
         forceDesktopMode: true,
       })
     },
+    async setCurrentPageAsBookPoster() {
+      let imageFile = await getFileFromUrl(this.currentPage.url, 'poster', 'image/jpeg', {credentials: 'include'})
+      this.resizeImageForPoster(imageFile, async (newImageFile: File) => {
+        await this.$komgaBooks.uploadThumbnail(this.book.id, newImageFile, true)
+        this.sendNotification(`${this.$t('bookreader.book_poster_set_to_current_page')}`)
+      })
+    },
+    async setCurrentPageAsSeriesPoster() {
+      let imageFile = await getFileFromUrl(this.currentPage.url, 'poster', 'image/jpeg', {credentials: 'include'})
+      this.resizeImageForPoster(imageFile, async (newImageFile: File) => {
+        await this.$komgaSeries.uploadThumbnail(this.series.id, newImageFile, true)
+        this.sendNotification(`${this.$t('bookreader.series_poster_set_to_current_page')}`)
+      })
+    },
+    resizeImageForPoster(imageFile: File, uploadCallback: (newImageFile: File) => void) {
```

Comment:
Seems like this could be pulled out of the component, and into a function file.
I'm also not a big fan of callbacks, i would prefer an async function that returns a `File`.